File: src/leetcode/2026/06/Solution1002.java
import java.util.List;

public class Solution1002 {
    public List<String> commonChars(String[] words) {
        int[] charCount = new int[26];
        for (char c : words[0].toCharArray()) {
            charCount[c - 'a']++;
        }

        for (int i = 1; i < words.length; i++) {
            int[] currentCount = new int[26];
            for (char c : words[i].toCharArray()) {
                currentCount[c - 'a']++;
            }
            for (int j = 0; j < 26; j++) {
                charCount[j] = Math.min(charCount[j], currentCount[j]);
            }
        }

        List<String> result = new java.util.ArrayList<>();
        for (int i = 0; i < 26; i++) {
            for (int j = 0; j < charCount[i]; j++) {
                result.add(String.valueOf((char) (i + 'a')));
            }
        }

        return result;
    }

    public static void main(String[] args) {
        Solution1002 solution = new Solution1002();

        String[] words1 = { "bella", "label", "roller" };
        System.out.println(solution.commonChars(words1)); // Output: ["e", "l", "l"]

        String[] words2 = { "cool", "lock", "cook" };
        System.out.println(solution.commonChars(words2)); // Output: ["c", "o"]
    }
}

File: src/leetcode/2026/06/Solution1002.cpp
#include <vector>
#include <string>
using namespace std;

class Solution1002 {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<string> res;
        vector<int> count(26, 0);
        for (char c : words[0]) {
            count[c - 'a']++;
        }

        for (int i = 1; i < words.size(); i++) {
            vector<int> temp(26, 0);
            for (char c : words[i]) {
                temp[c - 'a']++;
            }
            for (int j = 0; j < 26; j++) {
                count[j] = min(count[j], temp[j]);
            }
        }

        for (int i = 0; i < 26; i++) {
            while (count[i] > 0) {
                res.push_back(string(1, 'a' + i));
                count[i]--;
            }
        }
        return res;
    }
};

int main() {
    Solution1002 solution;

    vector<string> words = {"bella", "label", "roller"};
    vector<string> res = solution.commonChars(words);
    for (const string& s : res) {
        printf("%s ", s.c_str());
    }
    printf("\n");

    vector<string> words2 = {"cool", "lock", "cook"};
    vector<string> res2 = solution.commonChars(words2);
    for (const string& s : res2) {
        printf("%s ", s.c_str());
    }
    printf("\n");
    
    return 0;
}